### PR TITLE
ENH: consistent `Styler.highlight_X` arg signature with `props` 

### PR DESCRIPTION
**Pull request by attack68**
This splits #39821 into separate PRs.

The `See Also` doc strings here reference methods created in those PRs - if we push this through now I will fix it later if those methods don't get approved before 1.3.0.

The tests that were edited contained redundant code and were not testing.

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -1648,79 +1649,124 @@ def highlight_null(
...
+
         Returns
         -------
         self : Styler
+
+        See Also
+        --------
+        Styler.highlight_max: Highlight the maximum with a style.
+        Styler.highlight_min: Highlight the minimum with a style.
+        Styler.highlight_quantile: Highlight values defined by a quantile with a style.
```
Comment (jreback) [inline pandas/io/formats/style.py:1666]: need to take these new highlites out

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -1648,79 +1649,124 @@ def highlight_null(
...
+        Styler.highlight_between: Highlight a defined range with a style.
+
+        Notes
+        -----
+        Uses ``pandas.isna()`` to detect the missing values.
```
Comment (jreback) [inline pandas/io/formats/style.py:1671]: why is this here? we use isna for detecting missing values by default, so this is not surprising.

File: pandas/io/formats/style.py [hunk 3]
```diff
@@ -1648,79 +1649,124 @@ def highlight_null(
...
 
         Returns
         -------
         self : Styler
+
+        See Also
+        --------
+        Styler.highlight_null: Highlight missing values with a style.
+        Styler.highlight_max: Highlight the maximum with a style.
+        Styler.highlight_quantile: Highlight values defined by a quantile with a style.
```
Comment (jreback) [inline pandas/io/formats/style.py:1760]: same

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style.py]: removed

File: pandas/io/formats/style.py (hunk 2 above)
Comment (attack68) [reply pandas/io/formats/style.py]: removed

File: pandas/io/formats/style.py (hunk 3 above)
Comment (attack68) [reply pandas/io/formats/style.py]: removed

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. ping on greenish (or green if you had rebased recently)

**Comment by attack68** [comment]
@jreback green now

**Comment by jreback** [comment]
thanks @attack68
